internal/messaging/mint_v2.go

```diff
@@ -109,32 +110,34 @@ func (h *evmResponseHandler) processMintResponseV2(ctx context.Context, response
 	response.BuyTransactionId = receipt.TxHash.Hex()
 }
 
-func (h *evmResponseHandler) getPriceAndTokenV2(ctx context.Context, price *typesv2.Price) (*big.Int, common.Address, error) {
+func (h *evmResponseHandler) getPriceAndTokenV2(ctx context.Context, price *typesv2.Price) (*big.Int, common.Address, *big.Int, error) {
 	priceBigInt := big.NewInt(0)
 	paymentToken := zeroAddress
+	isoCurrency := big.NewInt(0)
 	var err error
 	switch currency := price.Currency.Currency.(type) {
 	case *typesv2.Currency_NativeToken:
 		priceBigInt, err = h.bookingService.ConvertPriceToBigInt(price.Value, price.Decimals, int32(18)) // CAM uses 18 decimals
 		if err != nil {
-			return nil, zeroAddress, fmt.Errorf("error minting NFT: %w", err)
+			return nil, zeroAddress, nil, fmt.Errorf("error minting NFT: %w", err)
 		}
 	case *typesv2.Currency_TokenCurrency:
 		if !common.IsHexAddress(currency.TokenCurrency.ContractAddress) {
-			return nil, zeroAddress, fmt.Errorf("invalid contract address: %s", currency.TokenCurrency.ContractAddress)
+			return nil, zeroAddress, nil, fmt.Errorf("invalid contract address: %s", currency.TokenCurrency.ContractAddress)
 		}
 		contractAddress := common.HexToAddress(currency.TokenCurrency.ContractAddress)
 		tokenDecimals, err := h.erc20.Decimals(ctx, contractAddress)
 		if err != nil {
-			return nil, zeroAddress, fmt.Errorf("failed to fetch token decimals: %w", err)
+			return nil, zeroAddress, nil, fmt.Errorf("failed to fetch token decimals: %w", err)
 		}
 		priceBigInt, err = h.bookingService.ConvertPriceToBigInt(price.Value, price.Decimals, tokenDecimals)
 		if err != nil {
-			return nil, zeroAddress, err
+			return nil, zeroAddress, nil, err
 		}
 		paymentToken = contractAddress
 	case *typesv2.Currency_IsoCurrency:
 		// For IsoCurrency, keep price as 0 and paymentToken as zeroAddress
+		isoCurrency = big.NewInt(int64(currency.IsoCurrency))
```

Comment:
For off chain payment [we decided](https://github.com/chain4travel/camino-messenger-contracts/pull/49) to use `address(1)` as payment token address.

```golang
paymentToken = common.HexToAddress("0x0000000000000000000000000000000000000001")
```

And also include the price in the price field. But we need to decide how many decimals we like have with the ISO currencies. @Noctunus ?

```golang
priceBigInt, err = h.bookingService.ConvertPriceToBigInt(price.Value, price.Decimals, int32(5))
```

Reply:
If we want to cover most of the countries 2 decimals
If we want to cover almost all countries (including Algeria, Bahrain, Iraq, Jordan, Kuwait, Libya and Tunisia) 3.
If we want to cover all currencies (including Chilean unidad de fomento) 4.

But to be fair, this could also just be an external definition as we don't do anything with the price except transporting it. Meaning we could also just define it as 0 and let the ISO currency be the off-chain definition of the decimal places.
For instance if you wanna send 100 dollar you have to specify 10000 as it defines 2 decimal places, if you wanna send 100 yen => 100 (0 decimals) and for Bahraini dinar it's 100000 (3 decimals).

So maybe we just skip the (static) decimals definition and let the currency decide, wdyt?

Reply:
Having it dynamic means that the bot should know each currency and it's decimals, no? Because we are getting it from the mint response as a Price type.

Reply:
We had a meeting (Ekrem, Sam, myself) about that and decided to go with a static definition of 6 decimal places for off-chain payment.

How that works: For on-chain we have the decimal definition in the ERC-20 of the on-chain currency defining the decimal places of that particular currency. 
We do the same transformation with off-chain currencies but instead of querying we just use a static definition of 6 decimal places for the target currency.
Meaning if for instance from the partner plugin a response arrives with off-chain payment of 600,43 USD (value 60043 decimals 2) we just convert it over to store it in the booking token to value: 600430000 
decimals: 6

The only thing which might have been an argument against having 6 static decimals is size but as it's stored in the Booking-Token SC in the value field anyways as a 256bit integer it won't negatively impact the size of the booking token.